src/irc.nim

```diff
@@ -298,14 +300,41 @@ proc connect*(irc: Irc) =
   assert(irc.port != Port(0))
 
   irc.status = SockConnecting
-  irc.sock.connect(irc.address, irc.port)
+  # check if proxyAddr has been provided
+  if irc.proxyAddr != "":
+    var p = "  "
+    p[0] = cast[char](irc.port.uint16 shr 8)
+    p[1] = cast[char](irc.port)
```

Comment:
You shouldn't need to cast here, can't you do a simple type conversion? It's much safer.

Reply:
trying type conversion for line 307 results in 

Error: unhandled exception: value out of range [RangeError]

casting seems to be the only way I can get it to work. Unless there's a way to convert it I'm not seeing?